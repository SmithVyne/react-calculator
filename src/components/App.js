import React from 'react';
import { Route, Switch, Link } from 'react-router-dom';
import Home from './Home';
import Calculator from './Calculator';
import Quote from './Quote';
import '../styles/App.css';

const App = () => (
  <>
    <nav>
      <h1 id="logo">Math Magicians</h1>
      <div id="nav-menu">
        <Link to="/" className="nav-links">Home</Link>
        |
        <Link to="/calculator" className="nav-links">Calculator</Link>
        |
        <Link to="/quote" className="nav-links">Quote</Link>
      </div>
    </nav>
    <main>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/calculator" component={Calculator} />
        <Route path="/quote" component={Quote} />
      </Switch>
    </main>
  </>
);

export default App;
