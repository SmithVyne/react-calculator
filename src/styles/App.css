nav {
  position: fixed;
  top: 0;
  left: 0;
  max-height: 80px;
  width: 100%;
  padding: 17px;
  background-color: #fff;
  display: flex;
  align-items: center;
  z-index: 1;
}

#logo {
  margin-right: auto;
  color: #000;
  font-size: 50px;
}

#nav-menu {
  display: flex;
  /* width: 205px; */
  /* min-width: 220px; */
  justify-content: space-between;
  font-size: 23px;
  color: black;
}

a.nav-links {
  color: rgb(64, 28, 165);
}
