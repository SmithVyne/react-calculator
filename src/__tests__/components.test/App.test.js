import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from '../../components/App';

describe('it displays the website', () => {
  it('it has a logo "MATH MAGICIANS"', () => {
    const { getByText } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    getByText('Math Magicians');
  });

  it('it has navigation links in the nav-bar', () => {
    const { getByRole, getByText } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    const nav = getByRole('navigation');
    const home = getByText('Home');
    const calculator = getByText('Calculator');
    const quote = getByText('Quote');

    expect(nav).toContainElement(home);
    expect(nav).toContainElement(calculator);
    expect(nav).toContainElement(quote);
  });

  it('it updates the dom when home is clicked', () => {
    const { getByRole, getByText } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );

    fireEvent.click(getByText('Home'));
    expect(getByRole('main')).toMatchSnapshot();
  });

  it('it updates the dom when calculator is clicked', () => {
    const { getByRole, getByText } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );

    fireEvent.click(getByText('Calculator'));
    expect(getByRole('main')).toMatchSnapshot();
  });
});
