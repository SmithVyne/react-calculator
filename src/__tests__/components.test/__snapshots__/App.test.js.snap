// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it displays the website it updates the dom when calculator is clicked 1`] = `
<main>
  <div
    class="calcuText"
  >
    Let's do some math!
  </div>
  <div
    class="app"
  >
    <div
      class="display"
    >
      0
    </div>
    <div
      class="buttonPanel"
    >
      <div
        class="topButtons"
      >
        <button
          class=""
          type="button"
        >
          AC
        </button>
        <button
          class=""
          type="button"
        >
          +/-
        </button>
        <button
          class=""
          type="button"
        >
          %
        </button>
        <button
          class="orangeBg"
          type="button"
        >
          รท
        </button>
        <button
          class=""
          type="button"
        >
          7
        </button>
        <button
          class=""
          type="button"
        >
          8
        </button>
        <button
          class=""
          type="button"
        >
          9
        </button>
        <button
          class="orangeBg"
          type="button"
        >
          x
        </button>
        <button
          class=""
          type="button"
        >
          4
        </button>
        <button
          class=""
          type="button"
        >
          5
        </button>
        <button
          class=""
          type="button"
        >
          6
        </button>
        <button
          class="orangeBg"
          type="button"
        >
          -
        </button>
        <button
          class=""
          type="button"
        >
          1
        </button>
        <button
          class=""
          type="button"
        >
          2
        </button>
        <button
          class=""
          type="button"
        >
          3
        </button>
        <button
          class="orangeBg"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="buttomButtons"
      >
        <button
          class=""
          type="button"
        >
          0
        </button>
        <button
          class=""
          type="button"
        >
          .
        </button>
        <button
          class="orangeBg"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</main>
`;

exports[`it displays the website it updates the dom when home is clicked 1`] = `
<main>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
  </p>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
  </p>
</main>
`;
